MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2653,26 +2653,121 @@ public string GetAudioFilterParam(EncodingJobInfo state, EncodingOptions encodin
+                        }
+
                         break;
-                    case DownMixStereoAlgorithms.NightmodeDialogue:
-                        filters.Add("pan=stereo|c0=c2+0.30*c0+0.30*c4|c1=c2+0.30*c1+0.30*c5");
+                    }
+
+                    case 4:
+                    {
+                        switch (encodingOptions.DownMixStereoAlgorithm)
+                        {
+                            case DownMixStereoAlgorithms.Rfc7845:
+                                filters.Add("pan=stereo|c0=0.422650*c0+0.366025*c2+0.211325*c3|c1=0.422650*c1+0.366025*c3+0.211325*c2");
+                                break;
+                            case DownMixStereoAlgorithms.Dave750:
+                            case DownMixStereoAlgorithms.NightmodeDialogue:
+                            case DownMixStereoAlgorithms.None:
+                            default:
+                                break;
+                        }
+
                         break;
-                    case DownMixStereoAlgorithms.None:
+                    }
+
+                    case 5:
+                    {
+                        switch (encodingOptions.DownMixStereoAlgorithm)
+                        {
+                            case DownMixStereoAlgorithms.Rfc7845:
+                                filters.Add("pan=stereo|c0=0.460186*c2+0.650802*c0+0.563611*c3+0.325401*c4|c1=0.460186*c2+0.650802*c1+0.563611*c4+0.325401*c3");
+                                break;
+                            case DownMixStereoAlgorithms.Dave750:
+                            case DownMixStereoAlgorithms.NightmodeDialogue:
+                            case DownMixStereoAlgorithms.None:
+                            default:
+                                break;
+                        }
+
+                        break;
+                    }
+
+                    case 6:
+                    {
+                        switch (encodingOptions.DownMixStereoAlgorithm)
+                        {
+                            case DownMixStereoAlgorithms.Dave750:
+                                filters.Add("pan=stereo|c0=0.5*c2+0.707*c0+0.707*c4+0.5*c3|c1=0.5*c2+0.707*c1+0.707*c5+0.5*c3");
+                                break;
+                            case DownMixStereoAlgorithms.NightmodeDialogue:
+                                filters.Add("pan=stereo|c0=c2+0.30*c0+0.30*c4|c1=c2+0.30*c1+0.30*c5");
+                                break;
+                            case DownMixStereoAlgorithms.Rfc7845:
+                                filters.Add("pan=stereo|c0=0.374107*c2+0.529067*c0+0.458186*c4+0.264534*c5+0.374107*c3|c1=0.374107*c2+0.529067*c1+0.458186*c5+0.264534*c4+0.374107*c3");
+                                break;
+                            case DownMixStereoAlgorithms.None:
+                            default:
+                                break;
+                        }
+
+                        break;
+                    }
```

Comment:
A longer example here. It reduces the nesting but I can't tell if readability suffers slightly.
```suggestion
                    case 6 when encodingOptions.DownMixStereoAlgorithm == DownMixStereoAlgorithms.Dave750:
                        filters.Add("pan=stereo|c0=0.5*c2+0.707*c0+0.707*c4+0.5*c3|c1=0.5*c2+0.707*c1+0.707*c5+0.5*c3");
                        break;
                    case 6 when encodingOptions.DownMixStereoAlgorithm == DownMixStereoAlgorithms.NightmodeDialogue:
                        filters.Add("pan=stereo|c0=c2+0.30*c0+0.30*c4|c1=c2+0.30*c1+0.30*c5");
                        break;
                    case 6 when encodingOptions.DownMixStereoAlgorithm == DownMixStereoAlgorithms.Rfc7845:
                        filters.Add("pan=stereo|c0=0.374107*c2+0.529067*c0+0.458186*c4+0.264534*c5+0.374107*c3|c1=0.374107*c2+0.529067*c1+0.458186*c5+0.264534*c4+0.374107*c3");
                        break;
                    default:
                        break;
```